Documentation/proposals/202212-scrape-config.md

```diff
@@ -0,0 +1,60 @@
+# ScrapeConfig CRD
+* Owners:
+  * [xiu](https://github.com/xiu)
+* Related Tickets:
+  * [#2787](https://github.com/prometheus-operator/prometheus-operator/issues/2787)
+  * [#3447](https://github.com/prometheus-operator/prometheus-operator/issues/3447)
+* Other docs:
+  * n/a
+
+This document aims at creating a lower level `ScrapeConfig` Custom Resource that defines additional scrape configurations the kubernetes way.
+
+# Why
+
+prometheus-operator misses a way to scrape external targets using CRD. Users have either been abusing the `Probe` CRD (#3447) or `additionalScrapeConfig` to do so.
```

Comment:
Thanks! What do you think about:
> prometheus-operator misses a way to scrape external targets using CRD. Users have either been abusing the `Probe` CRD (#3447) or `additionalScrapeConfig` to do so. 
> We currently have several CRDs generating scrape configurations. With this proposition of creating a new `ScrapeConfig` CRD, we also have an interest in consolidating the logic to generate scrape configuration in one place by having all the other CRDs generate `ScrapeConfig` resources instead.

Reply:
Changed it slightly but feel free to use which one of the texts looks better to you :) 

> prometheus-operator misses a way to scrape external targets using CRD. Users have either been abusing the Probe CRD (https://github.com/prometheus-operator/prometheus-operator/issues/3447) or additionalScrapeConfig to do so.
Furthermore, currently, there is a lot of code duplication due to the operator supporting several CRDs that generate scrape configurations. With the new ScrapeConfig CRD, it would be possible to consolidate some of that logic, where the other `*Monitor` CRDs could be migrated so that they create a ScrapeConfig resource that would ultimately be used by the operator to generate scrape configuration.